src/EntityBuilder.ts

```diff
@@ -1,78 +1,88 @@
-export class EntityBuilder {
+import { Entity, PartialEntityJson } from './Entity';
+import { Buildable, Constructor } from './support/Type';
+import { TypeMetadata } from './support/metadata/TypeMetadata';
+import { defaultMetadataStorage } from './support/storage';
+import { isEntityType } from './support/isEntityType';
+import { StringHelper } from './support/StringHelper';
 
+export class EntityBuilder {
     public static enableCamelConversion = true;
 
-    /**
-     * Build an entity object from source data.
-     *
-     * @param buildClass
-     * @param sourceData
-     * @returns {any}
-     */
-    public static buildOne<T>(buildClass: any, sourceData: Object): T | any {
-        this.checkClassValidity(buildClass);
+    public static async buildOne<T extends Entity>(
```

Comment:
I think making all these methods async reduces the library's usability. Is there a way to keep it synchronous? This kind of use greatly reduces ease-of-use as it nearly mandates use of `async` functions, and I don't think people should have to deal with promises to convert POJOs to typed classes.

Reply:
Asynchronousity is needed for the deferred resolution I introduced last year. I agree that it's not a great DX but having both synchronous *and* asynchronous build methods makes the codebase quite complicated with the type resolutions introduced in this MR, and it's also not a great DX either.

I'll see if I can remove asynchronous builds altogether by creating different constraints for deferred resolutions.